File: common/src/main/java/dev/number6/db/dagger/DynamoDBMapperModule.java
package dev.number6.db.dagger;

import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClientBuilder;
import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
import dagger.Module;
import dagger.Provides;

@Module
public class DynamoDBMapperModule {

    @Provides
    public DynamoDBMapper dbMapper() {
        AmazonDynamoDB client = AmazonDynamoDBClientBuilder.defaultClient();
        return new DynamoDBMapper(client);
    }
}

File: common/src/main/kotlin/dev/number6/db/dagger/DynamoDBMapperModule.kt
package dev.number6.db.dagger

import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClientBuilder
import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper
import dagger.Module
import dagger.Provides

@Module
class DynamoDBMapperModule {
    @Provides
    fun dbMapper(): DynamoDBMapper {
        val client = AmazonDynamoDBClientBuilder.defaultClient()
        return DynamoDBMapper(client)
    }
}